src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -766,19 +766,69 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
+            for (const auto& group_name : group_names) {
+                const auto& group = this->snapshots.back().groups.get(group_name);
+                const auto& parent_name = group.parent();
+                if (!parent_name.empty())
+                {
+                    const std::string& downtree_node = group_name;
+                    const std::string& uptree_node = parent_name;
+                    if (vfp_table == 0) {
+                        // If vfp table is defaulted (or set to 0) then the group is not part of the network
+                        if (network.has_node(downtree_node) && network.has_node(uptree_node)) 
+                            // If it was part of the network then drop it
+                            network.drop_branch(uptree_node, downtree_node);
+                    } else {
+                         const auto alq_eq = Network::Branch::AlqEqfromString(record.getItem<ParserKeywords::GRUPNET::ALQ_SURFACE_DENSITY>().get<std::string>(0));
+                         if (alq_eq == Network::Branch::AlqEQ::ALQ_INPUT) {
+                             const double alq_value = record.getItem<ParserKeywords::GRUPNET::ALQ>().get<double>(0);
+                             network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_value));
+                        } else {
+                             network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_eq));
+                        }
+                    }
+                }
+                bool is_valid_node = false;
+                Network::Node node { group_name };
+                const bool is_terminal_node = pressure_item.hasValue(0) && (pressure_item.get<double>(0) >= 0);
+                if (is_terminal_node) {
+                    if (vfp_table == 0) {
+                        is_valid_node = true;
+                        node.terminal_pressure(pressure_item.getSIDouble(0));
+                    } else {
+                         std::string msg = fmt::format("The group {} is a terminal node of the network and should not have a vfp table assigned to it.", group_name);
+                         throw OpmInputError(msg, handlerContext.keyword.location());
+                    }
+                } else {
+                     if (vfp_table != 0){
+                         // Only non-terminal nodes with non-default vfp table value can be part of the network
+                         is_valid_node = true;
+                     } else {
+                          if(network.has_node(node.name())) {
+                             std::string msg = fmt::format("The group {} is not a terminal node of the network and should have a vfp table assigned to it.", group_name);
+                             throw OpmInputError(msg, handlerContext.keyword.location());
+                          }
+                    }
+                }
+                if (is_valid_node)
+                    nodes.push_back(node);
```

Comment:
not sure we need to have a vector to store the `nodes` first. We can just `update_node` to the network.

Reply:
The `update_node` assumes that the node is already associated to a branch therefore to avoid  "dangling node" throws when calling `update_node`  the nodes are collected in a vector and after the branches are created the update of nodes is done.

Reply:
Yes, you are right. There is some improvements can be done in the update_node function. For example, if the first record of GRUPNET is a terminal node, then there is no branch in the network yet, it will not belong to any branch. 

I think these parts of the code will evolve with the network development. We do not have to correct this in this PR.